pkg/reconciler/v1alpha1/taskrun/resources/image_exporter.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"encoding/json"
+	"flag"
+	"os"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/names"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	imageDigestExporterImage = flag.String("imagedigest-exporter-image", "override-with-imagedigest-exporter-image:latest", "The container image containing our image digest exporter binary.")
+)
+
+// AddOutputImageDigestExporter add a step to check the index.json for all output images
+func AddOutputImageDigestExporter(
+	tr *v1alpha1.TaskRun,
+	taskSpec *v1alpha1.TaskSpec,
+	gr GetResource,
+	logger *zap.SugaredLogger,
+) error {
+
+	output := []*v1alpha1.ImageResource{}
+	if len(tr.Spec.Outputs.Resources) > 0 {
+		for _, trb := range tr.Spec.Outputs.Resources {
+			boundResource, err := getBoundResource(trb.Name, tr.Spec.Outputs.Resources)
+			if err != nil {
+				logger.Errorf("Failed to get bound resource: %s", err)
+				return err
+			}
+
+			resource, err := getResource(boundResource, gr)
+			if err != nil {
+				logger.Errorf("Failed to get output pipeline Resource for taskRun %q resource %v; error: %s", tr.Name, boundResource, err.Error())
+				return err
+			}
+			if resource.Spec.Type == v1alpha1.PipelineResourceTypeImage {
+				imageResource, err := v1alpha1.NewImageResource(resource)
+				if err != nil {
+					logger.Errorf("Invalid Image Resource for taskRun %q resource %v; error: %s", tr.Name, boundResource, err.Error())
+					return err
+				}
+				for _, o := range taskSpec.Outputs.Resources {
+					if o.Name == boundResource.Name {
+						if o.OutputImagePath != "" {
+							if _, err := os.Stat(o.OutputImagePath); os.IsNotExist(err) {
+								if err := os.MkdirAll(o.OutputImagePath, os.ModePerm); err != nil {
+									return err
```

Comment:
im a bit confused by this part, is this making a directory on the host running the controller itself? if so, why? (maybe some comments would clear it up?)

Reply:
sorry can't actually remember why I added that, but you are correct, doesn't make sense, will remove it.